packages/nuka/src/carousel.tsx

```diff
@@ -18,672 +18,682 @@ import {
 } from './utils';
 import { useFrameHeight } from './hooks/use-frame-height';
 import { getDotIndexes } from './default-controls';
-
-export const Carousel = (rawProps: CarouselProps): React.ReactElement => {
-  /**
-   * We need this cast because we want the component's properties to seem
-   * optional to external users, but always-present for the internal
-   * implementation.
-   *
-   * This cast is safe due to the `Carousel.defaultProps = defaultProps;`
-   * statement below. That guarantees all the properties are present, since
-   * `defaultProps` has type `InternalCarouselProps`.
-   */
-  const props = rawProps as InternalCarouselProps;
-
-  const {
-    adaptiveHeight,
-    adaptiveHeightAnimation,
-    afterSlide,
-    animation,
-    autoplay,
-    autoplayInterval,
-    autoplayReverse,
-    beforeSlide,
-    cellAlign,
-    cellSpacing,
-    children,
-    className,
-    disableAnimation,
-    dragging: desktopDraggingEnabled,
-    dragThreshold: propsDragThreshold,
-    enableKeyboardControls,
-    frameAriaLabel,
-    innerRef,
-    keyCodeConfig,
-    onDrag,
-    onDragEnd,
-    onDragStart,
-    onUserNavigation,
-    pauseOnHover,
-    renderAnnounceSlideMessage,
-    scrollMode,
-    slideIndex,
-    slidesToScroll: propsSlidesToScroll,
-    slidesToShow,
-    speed,
-    style,
-    swiping: mobileDraggingEnabled,
-    wrapAround,
-    zoomScale,
-  } = props;
-
-  const filteredSlides = React.Children.toArray(children).filter(Boolean);
-  const slideCount = filteredSlides.length;
-
-  const slidesToScroll =
-    animation === 'fade' ? slidesToShow : propsSlidesToScroll;
-
-  const [currentSlide, setCurrentSlide] = useState<number>(() =>
-    getDefaultSlideIndex(
+import { useForwardRef } from './hooks/use-forward-ref';
+
+// eslint-disable-next-line react/display-name
```

Comment:
You can avoid having to disable this rule by naming the function argument like `function Carousel(rawProps, ref): React.ReactElement {`